fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/expressions/FunctionKeyExpression.java

```diff
@@ -371,4 +371,18 @@ public static Registry instance() {
             return functions;
         }
     }
+
+    @Override
+    public boolean equivalentForSort(@Nonnull KeyExpression other) {
+        if (getClass() != other.getClass()) {
+            return false;
+        }
+
+        FunctionKeyExpression that = (FunctionKeyExpression) other;
+        if (!this.getName().equals(that.getName())) {
+            return false;
+        }
+
+        return this.getArguments().equivalentForSort(that.getArguments());
+    }
```

Comment:
It seems like as we get more equals methods here with slightly different semantics, I wonder if it might be worth it to abstract this out slightly differently. Like, I think the only difference here between this and the other `equals` method is whether `.equals` or `.equivalentForSort` is called. It almost seems like they should all call an `equalsWithoutChildren` method and then different implementations can check the children for equality, or something.

@nschiefer, do you have thoughts (especially with `equalsWithoutChildren` coming in #634)?

Reply:
I'm not sure that sharing the code would actually be clearer, since the leaf would need to be different.

Reply:
I still think it would be nice to normalize these at some point, though I agree some care would be taken to make it not awkward. I filed #678 about this, as I don't think it needs to block this PR.